src/plugins/data_source/server/client/configure_client.ts

```diff
@@ -117,52 +161,75 @@ export const getCredential = async (
  * @returns child client.
  */
 const getQueryClient = async (
-  rootClient: Client,
-  dataSource: DataSourceAttributes,
+  dataSourceAttr: DataSourceAttributes,
+  { addClientToPool }: OpenSearchClientPoolSetup,
```

Comment:
if we just need this function, can we just pass it instead of `OpenSearchClientPoolSetup` object?

Reply:
meanwhile, which pool it expects? the legacy client pool or the opensearch client pool?

Reply:
it's expecting opensearch client pool. For legacy client, there's another `getQueryClient` dedicated to legacy client use case , see `configure_legacy_client`
https://github.com/zhongnansu/OpenSearch-Dashboards/blob/dafc29decb5e4102a08866861e744efc9e1ef61d/src/plugins/data_source/server/legacy/configure_legacy_client.ts#L66